map/src/menu/login/ChangeResetPwd.jsx

```diff
@@ -0,0 +1,294 @@
+import headerStyles from '../trackfavmenu.module.css';
+import {
+    AppBar,
+    Box,
+    Button,
+    Icon,
+    IconButton,
+    InputAdornment,
+    ListItemText,
+    TextField,
+    Toolbar,
+    Typography,
+} from '@mui/material';
+import styles from './login.module.css';
+import errorStyles from './../errors/errors.module.css';
+import { closeHeader } from '../actions/HeaderHelper';
+import React, { useContext, useEffect, useState } from 'react';
+import { ReactComponent as CloseIcon } from '../../assets/icons/ic_action_close.svg';
+import { ReactComponent as UserPasswordIcon } from '../../assets/icons/ic_action_user_password.svg';
+import AppContext from '../../context/AppContext';
+import { closeLoginMenu, EMPTY_INPUT, openLogin } from '../../manager/LoginManager';
+import { useTranslation } from 'react-i18next';
+import { userActivate, userRegisterAndSendCode } from '../../manager/AccountManager';
+import i18n from 'i18next';
+import { formatString } from '../../manager/SettingsManager';
+import loginStyles from './login.module.css';
+import { useNavigate } from 'react-router-dom';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+
+export default function ChangeResetPwd() {
+    const ctx = useContext(AppContext);
+
+    const { t } = useTranslation();
+    const navigate = useNavigate();
+    const lang = i18n.language;
+
+    const [userEmail, setUserEmail] = useState(EMPTY_INPUT);
+    const [emailError, setEmailError] = useState(EMPTY_INPUT);
+    const [passwordError, setPasswordError] = useState(EMPTY_INPUT);
+    const [userPassword1, setUserPassword1] = useState(EMPTY_INPUT);
+    const [userPassword2, setUserPassword2] = useState(EMPTY_INPUT);
+    const [openCodeInput, setOpenCodeInput] = useState(false);
+    const [code, setCode] = useState(EMPTY_INPUT);
+    const [resetPasswordError, setResetPasswordError] = useState(EMPTY_INPUT);
+    const [openResetStatus, setOpenResetStatus] = useState(false);
+    const [showPassword1, setShowPassword1] = useState(false);
+    const [showPassword2, setShowPassword2] = useState(false);
+
+    const handleEmailChange = (e) => {
+        if (emailError !== EMPTY_INPUT) {
+            setEmailError(EMPTY_INPUT);
+        }
+        setUserEmail(e.target.value);
+    };
+
+    const handleCodeChange = (e) => {
+        setCode(e.target.value);
+    };
+
+    const handlePassword1Change = (e) => {
+        setUserPassword1(e.target.value);
+    };
+
+    const handlePassword2Change = (e) => {
+        setUserPassword2(e.target.value);
+    };
+
+    const handleClickShowPassword1 = () => {
+        setShowPassword1(!showPassword1);
+    };
+
+    const handleMouseDownPassword1 = (event) => {
+        event.preventDefault();
+    };
+
+    const handleClickShowPassword2 = () => {
+        setShowPassword2(!showPassword2);
+    };
+
+    const handleMouseDownPassword2 = (event) => {
+        event.preventDefault();
+    };
```

Comment:
please use 1-liners even without { } for the real one-liners:

const handleCodeChange = (e) => setCode(e.target.value);